apps/parsley/src/hooks/useSections/utils.ts

```diff
@@ -0,0 +1,97 @@
+import { Range } from "types/logs";
+import { trimSeverity } from "utils/string";
+
+interface SectionLineMetadata {
+  functionName: string;
+  status: "Running" | "Finished";
+}
+/**
+ * Parses a log line to extract metadata about a section entry.
+ * @param str - The log line to be processed.
+ * @returns The metadata about the section entry
+ * or null if the input line does not indicate the start or end of a section.
+ */
+export const processLine = (str: string): SectionLineMetadata | null => {
+  const regex = /(Running|Finished) command '[^']+' in function '([^']+)'.*/;
```

Comment:
For the `Running` command, why don't we check that it's `step 1` or `step 1.1`? It could help simplify some of the logic below